runtime/libc/cortex_m_backing.c

```diff
@@ -4,6 +4,41 @@ int printf_(const char* format, ...);
 
 volatile int CORTEX_M_ARG_C = 0;
 
+int runtime_main(int argc, char** argv) {
+    // Setup the linear memory and function table
+    alloc_linear_memory();
+    populate_table();
+    // Setup our allocation logic
+    runtime_heap_base = wasmg___heap_base;
+    printf("starting rhb %d\n", runtime_heap_base);
+    if (runtime_heap_base == 0) {
+        runtime_heap_base = memory_size;
+    }
```

Comment:
Is this how CM worked before? Jesus

How does this even work before `populate_globals` runs?

Reply:
Not sure! I haven't yet gotten familiar with `populate_globals`, and a cursory glance at `insert_runtime_globals` goes over my head. 🤣😅 Before this PR, the entry point of all three variants' (wasmception, wasi-sdk, and cortex-m) called `runtime_main`, which had this same order of instructions. So all of the variants appear to have worked this way, not just cortex m. I think that this was logged via a printf like `printf("starting rhb %d\n", runtime_heap_base);`. I think I've only see this log out `starting rhb 0`. Not sure what that means.

Reply:
Based on your comment, I'll move this into the `runtime_init` function after the calls to populate the table, globals, and memory. Sounds like there might be more to address here later.